core/domain/exp_domain.py

```diff
@@ -3526,29 +3528,60 @@ def _convert_states_v21_dict_to_v22_dict(cls, states_dict):
         """
         for key, state_dict in states_dict.iteritems():
             states_dict[key] = State.convert_html_fields_in_state(
-                state_dict, html_cleaner.convert_to_textangular)
+                state_dict, html_cleaner.convert_to_textangular, 'unused')
         return states_dict
 
     @classmethod
     def _convert_states_v22_dict_to_v23_dict(cls, states_dict):
         """Converts from version 22 to 23. Version 23 ensures that all
         all oppia-noninteractive-image tags have caption attribute.
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        for key, state_dict in states_dict.iteritems():
+            states_dict[key] = State.convert_html_fields_in_state(
+                state_dict, html_cleaner.add_caption_attr_to_image, 'unused')
+        return states_dict
+
+    @classmethod
+    def _convert_states_v23_dict_to_v24_dict(cls, states_dict, exp_id):
+        """Converts from version 23 to 24. Version 24 adds the dimensions of
+        images in the oppia-noninteractive-image tags.
+
         Args:
             states_dict: dict. A dict where each key-value pair represents,
                 respectively, a state name and a dict used to initialize a
                 State domain object.
+            exp_id: str. ID of the exploration.
 
         Returns:
             dict. The converted states_dict.
         """
         for key, state_dict in states_dict.iteritems():
             states_dict[key] = State.convert_html_fields_in_state(
-                state_dict, html_cleaner.add_caption_attr_to_image)
+                state_dict,
+                html_cleaner.add_dimensions_to_noninteractive_image_tag,
+                exp_id)
+            if state_dict['interaction']['id'] == 'ImageClickInput':
+                filename = state_dict['interaction']['customization_args'][
+                    'imageAndRegions']['value']['imagePath']
+                state_dict['interaction']['customization_args'][
+                    'imageAndRegions']['value']['imagePath'] = (
+                        html_cleaner.get_filepath_of_object_image(
+                            filename, exp_id))
+
         return states_dict
 
     @classmethod
     def update_states_from_model(
-            cls, versioned_exploration_states, current_states_schema_version):
+            cls, versioned_exploration_states, current_states_schema_version,
+            exploration_id):
```

Comment:
Maybe use the default argument here too?

Reply:
What default argument? I think it is okay to send the exploration_id along with it.

Reply:
I mean use the same approach as in `convert_html_fields_in_state`

Reply:
I think we don't need to that because the call for the function is same for every schema migration. (called only from one place in the loop where we migrate the exploration states from its version to the latest)